commit 34d211a2d5df4984a35b18d8ccacbe1d10abb067
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Wed Mar 16 08:04:07 2011 -0700

    Increase OSF partition limit from 8 to 18
    
    It turns out that while a maximum of 8 partitions may be what people
    "should" have had, you can actually fit up to 18 entries(*) in a sector.
    
    And some people clearly were taking advantage of that, like Michael
    Cree, who had ten partitions on one of his OSF disks.
    
    (*) The OSF partition data starts at byte offset 64 in the first sector,
        and the array of 16-byte partition entries start at offset 148 in
        the on-disk partition structure.
    
    Reported-by: Michael Cree <mcree@orcon.net.nz>
    Cc: stable@kernel.org (v2.6.38)
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    [dannf: Adjusted to apply to Debian's 2.6.32]

diff -urpN linux-source-2.6.32.orig/fs/partitions/osf.c linux-source-2.6.32/fs/partitions/osf.c
--- linux-source-2.6.32.orig/fs/partitions/osf.c	2011-03-22 23:27:01.507715211 -0600
+++ linux-source-2.6.32/fs/partitions/osf.c	2011-03-22 23:30:09.964362350 -0600
@@ -10,7 +10,7 @@
 #include "check.h"
 #include "osf.h"
 
-#define MAX_OSF_PARTITIONS 8
+#define MAX_OSF_PARTITIONS 18
 
 int osf_partition(struct parsed_partitions *state, struct block_device *bdev)
 {
